packages/emails/lib/getICalUID.ts

```diff
@@ -0,0 +1,36 @@
+import short from "short-uuid";
+import { v5 as uuidv5 } from "uuid";
+
+import { APP_NAME } from "@calcom/lib/constants";
+
+/**
+ * This function returns the iCalUID if a uid is passed or if it is present in the event that is passed
+ * @param uid - the uid of the event
+ * @param event - an event that already has an iCalUID or one that has a uid
+ * @param defaultToEventUid - if true, will default to the event.uid if present
+ *
+ * @returns the iCalUID whether already present or generated
+ */
+const getICalUID = ({
+  uid,
+  event,
+  defaultToEventUid,
+}: {
+  uid?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  event?: { iCalUID?: string | null; uid?: string | null; [key: string]: any };
+  defaultToEventUid?: boolean;
+}) => {
+  if (event?.iCalUID) return event.iCalUID;
+
+  if (defaultToEventUid && event?.uid) return `${event.uid}@${APP_NAME}`;
+
+  if (uid) return `${uid}@${APP_NAME}`;
+
+  const translator = short();
+
+  uid = translator.fromUUID(uuidv5(APP_NAME, uuidv5.URL));
+  return `${uid}@${APP_NAME}`;
```

Comment:
I'm a big fan of this PR; but this may break badly (I believe an iCAL uid HAS to be a 'valid' email address, whether it exists or not. But I think it's not a given that `APP_NAME` is actually a domain like `Cal.com`.

Reply:
It doesn't have to be a valid email address. It is recommended and I think it would be cool to have some of our branding for people that dig into this stuff.

Outlook's iCalUID it generates does not have a domain.

https://icalendar.org/New-Properties-for-iCalendar-RFC-7986/5-3-uid-property.html